sechub-wrapper-prepare/src/test/java/com/mercedesbenz/sechub/wrapper/prepare/modules/PrepareWrapperModuleSkopeoTest.java

```diff
@@ -0,0 +1,320 @@
+        /* prepare */
+        PrepareWrapperContext context = createContextEmptyConfig();
+        List<SecHubRemoteDataConfiguration> remoteDataConfigurationList = new ArrayList<>();
+        SecHubRemoteDataConfiguration remoteDataConfiguration = new SecHubRemoteDataConfiguration();
+        SecHubRemoteCredentialConfiguration credentials = new SecHubRemoteCredentialConfiguration();
+        SecHubRemoteCredentialUserData user = new SecHubRemoteCredentialUserData();
+        user.setName("my-example-name");
+        credentials.setUser(user);
+        remoteDataConfiguration.setCredentials(credentials);
+        remoteDataConfiguration.setLocation("my-example-location");
+        remoteDataConfiguration.setType("git");
+        remoteDataConfigurationList.add(remoteDataConfiguration);
+        context.setRemoteDataConfigurationList(remoteDataConfigurationList);
+
+        /* execute */
+        IllegalStateException exception = assertThrows(IllegalStateException.class, () -> moduleToTest.prepare(context));
+
+        /* test */
+        assertTrue(exception.getMessage().contains("Defined password must not be null or empty."));
+    }
+
+    @Test
+    void prepare_successful_when_user_credentials_are_configured_correctly() throws IOException {
+        /* prepare */
+        File tempDir = Files.createTempDirectory("upload-folder").toFile();
+        tempDir.deleteOnExit();
+        String filename = "testimage.tar";
+        writer.save(new File(tempDir, filename), "some text", true);
+
+        PrepareWrapperEnvironment environment = mock(PrepareWrapperEnvironment.class);
+        when(environment.getPdsPrepareUploadFolderDirectory()).thenReturn(tempDir.toString());
+        PrepareWrapperContext context = new PrepareWrapperContext(createFromJSON("{}"), environment);
+
+        List<SecHubRemoteDataConfiguration> remoteDataConfigurationList = new ArrayList<>();
+        SecHubRemoteDataConfiguration remoteDataConfiguration = new SecHubRemoteDataConfiguration();
+        SecHubRemoteCredentialConfiguration credentials = new SecHubRemoteCredentialConfiguration();
+        SecHubRemoteCredentialUserData user = new SecHubRemoteCredentialUserData();
+        user.setName("my-example-name");
+        user.setPassword("ghp_exampleAPITOKEN8ffne3l6g9f393r8fbcsf");
+        credentials.setUser(user);
+        remoteDataConfiguration.setCredentials(credentials);
+        remoteDataConfiguration.setLocation("my-example-location");
+        remoteDataConfiguration.setType("docker");
+        remoteDataConfigurationList.add(remoteDataConfiguration);
+        context.setRemoteDataConfigurationList(remoteDataConfigurationList);
+
+        /* execute */
+        moduleToTest.prepare(context);
+
+        /* test */
+        verify(skopeo).download(any(SkopeoContext.class));
+        verify(skopeo).cleanUploadDirectory(tempDir.toString());
+    }
+
+    @Test
+    void prepare_successful_when_no_credentials_are_configured() throws IOException {
```

Comment:
The test indicates no credentials are configured, but there are credentials configured during the test. Please change the `prepare` step of the test.